provisioning/device/src/Transports/Mqtt/ProvisioningTransportHandlerMqtt.cs

```diff
@@ -407,7 +424,7 @@ private MqttClientOptionsBuilder CreateMqttClientOptions(ProvisioningTransportRe
         private bool CertificateValidationHandler(MqttClientCertificateValidationEventArgs args)
         {
             return _settings.RemoteCertificateValidationCallback.Invoke(
-                new object(), //TODO Tim to check with Abhipsa about this and if it is necessary
+                new object(), // TODO: Tim to check with Abhipsa about this and if it is necessary
```

Comment:
For context, the object that is "supposed" to be used as the argument here is the "sender class" such as the http client, or the mqtt client here.

It looks like previews/v2 already has this todo resolved by passing in the mqtt client this layer uses. You may just need to pull the latest

Reply:
Oh, scratch that. This is only resolved in the IoT hub device client MQTT layer, not the DPS device client layer

Reply:
Here's what we did in the IoT hub device client: https://github.com/Azure/azure-iot-sdk-csharp/blob/eb20ec22df8b6c19ae483fe3277b81c63077333f/iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs#L827

Reply:
I believe this was only still TODO because it would require making the MQTT client a class level variable which would complicate things. At this point, it's just a question of if providing the MQTT client in this callback is of any value, or if we can get by with simpler code in our SDK without doing so.

Reply:
I suspect we would just need to make this callback defined in-line alongside the mqtt client instead of in a class level method. I'll take a stab at this later, so feel free to disregard this for now

Reply:
Updated

Reply:
https://github.com/Azure/azure-iot-sdk-csharp/pull/3005